src/components/input/index.js

```diff
@@ -14,38 +14,70 @@ export const Input = ({
   onWheel = () => null,
   disabled,
   value,
+  children,
+  defaultValue,
   pattern,
   onKeyPress,
   autoFocus = false,
 }) => (
   <div className={styles.container}>
-    <label>
+    <label htmlFor={name}>
+      <p>{label}</p>
       <input
         type={type}
         placeholder={placeholder}
-        name={name}
+        name={name || label}
         min={min}
         max={max}
         maxLength={maxlength}
-        defaultValue={value}
+        defaultValue={defaultValue === null ? '' : defaultValue}
```

Comment:
out of interest, why is defaultValue needed?

Reply:
This is one of the things I had the most issue with.
Basically it's the controlled vs uncontrolled input thing I linked in b23cd16ad679bdad05fe7090baca10c3f6146ab5 since now, the Input component was only exposing `value` but was used as `defaultValue`... So I unshadowed that and kept using uncontrolled/controlled inputs where needed, it would have been easier to use a form library as you suggested 😅